tests/e2e/error-boundary.test.ts

```diff
@@ -1,8 +1,17 @@
+import { type Request } from '@playwright/test'
 import { expect, test } from '#tests/playwright-utils.ts'
 
 test('Test root error boundary caught', async ({ page }) => {
-	await page.goto('/does-not-exist')
+	const pageUrl = '/does-not-exist'
+	await page.goto(pageUrl)
 
 	await expect(page.getByText(/We can't find this page/i)).toBeVisible()
-	// TODO: figure out how to assert the 404 status code
+
+	const listener = (request: Request) => {
+		if (request.url().includes(pageUrl)) {
+			request.response().then(response => expect(response?.status()).toBe(404))
```

Comment:
With assertions that happen asynchronously like this I always like to make certain that they can fail. I'm concerned that as it's currently written, playwright will think the test is over before this assertion runs. Can you verify that the test can fail? https://kentcdodds.com/blog/make-your-test-fail